### PR TITLE
Fixed Issue 26389

### PR DESCRIPTION
**Pull request by imrandotftr**
### Description (*)

Watermark is breaking product page when size is not set the issue fixed
### Fixed Issues (if relevant)

1. magento/magento2#26389: Watermark is breaking product page when size is not set

### Manual testing scenarios (*)

1.     Navigate to theme configuration in Content -> Design -> Configuration -> <some theme> -> Product Image Watermarks
2.     Upload a new watermark. Specify opacity and DO NOT specify Image size

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@imrandotftr Please accept the CLA and fix Static tests:
```
PHP Code Sniffer detected 2 violation(s): 

FILE: ...l/app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 10 WARNINGS AFFECTING 8 LINES
----------------------------------------------------------------------
 134 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 16 spaces but found 20
 135 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 16 spaces but found 20
 136 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 16 spaces but found 20
 137 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 12 spaces but found 16
 139 | WARNING | [ ] Expected "if (...) {\n"; found "if(...){\n"
 139 | WARNING | [x] Expected 1 space(s) after IF keyword; 0 found
 139 | WARNING | [x] Expected 1 space(s) after closing parenthesis;
     |         |     found 0
 141 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 20 spaces but found 28
 142 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 20 spaces but found 28
 147 | WARNING | [x] Multi-line function call not indented correctly;
     |         |     expected 16 spaces but found 24
----------------------------------------------------------------------
PHPCBF CAN FIX THE 9 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------



Failed asserting that 2 matches expected 0.
```

Don't worry about Functional Tests. Fix will be merged to `2.4-develop` soon for that issue.

File: app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php [hunk 1]
```diff
@@ -130,14 +130,22 @@ private function getWatermark(string $type, int $scopeId = null): array
...
 
         if ($file) {
-            $size = explode(
-                'x',
-                (string) $this->scopeConfig->getValue(
+            $watermarkSize = (string) $this->scopeConfig->getValue(
                     "design/watermark/{$type}_size",
                     ScopeInterface::SCOPE_STORE,
                     $scopeId
-                )
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php:139]: Extract to private method `getWatermarkSize(string $type): string`

File: app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php [hunk 2]
```diff
@@ -130,14 +130,22 @@ private function getWatermark(string $type, int $scopeId = null): array
...
+                );
+            $size = [];
+            if($watermarkSize != ''){
+                $size = explode(
+                            'x',
+                            (string) $this->scopeConfig->getValue(
+                                "design/watermark/{$type}_size",
+                                ScopeInterface::SCOPE_STORE,
+                                $scopeId
+                            )
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php:146]: You already fetched that value - why don't you use `$watermarkSize`?

File: app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php [hunk 3]
```diff
@@ -130,14 +130,22 @@ private function getWatermark(string $type, int $scopeId = null): array
...
                     "design/watermark/{$type}_size",
                     ScopeInterface::SCOPE_STORE,
                     $scopeId
-                )
-            );
+                );
+            $size = [];
+            if($watermarkSize != ''){
+                $size = explode(
+                            'x',
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Model/Product/Image/ParamsBuilder.php:141]: Extract to `const` - `private const WATERMARK_SIZE_SEPARATOR`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please find review comments.

**Comment by magento-engcom-team** [comment]
Hi @imrandotftr

Unfortunately, we cannot accept Pull Requests from branches created in `magento2` repository.
Please use your own fork as a base for this Pull Request.

Thank you.

**Pull request closed without merging**